src/sentry/snuba/metrics/query_builder.py

```diff
@@ -441,18 +417,27 @@ def _build_orderby(self) -> Optional[List[OrderBy]]:
     def __build_totals_and_series_queries(
         self, entity, select, where, groupby, orderby, limit, offset, rollup, intervals_len
     ):
-        # In a series query, we also need to factor in the len of the intervals
-        # array. The number of totals should never get so large that the
-        # intervals exceed MAX_POINTS, however at least a single group.
-        max_totals = max(MAX_POINTS // intervals_len, 1)
-        totals_limit = max_totals
-        if limit and limit.limit < totals_limit:
+        series_limit = None
+        if self._query_definition.include_series:
+            # In a series query, we also need to factor in the len of the intervals
+            # array. The number of totals should never get so large that the
+            # intervals exceed MAX_POINTS, however at least a single group.
+            totals_limit = max(MAX_POINTS // intervals_len, 1)
+            if limit and limit.limit < totals_limit:
+                totals_limit = limit.limit
+
+            # We want to make sure that the totals limit when multiplied byt intervals len always
+            # yields a number that is less than max points to ensure that we get full series of a
+            # group and it is not cut off because it exceeds snuba limit
+            while totals_limit * intervals_len > MAX_POINTS:
+                totals_limit -= 1
```

Comment:
Isn't this already guaranteed by line 425?

Reply:
Oh yeah you are right .. I got thrown off by the condition that was there in the previous commit
This is redundant. Will remove